components/lark-parser/src/syntax/member.rs

```diff
@@ -56,6 +86,67 @@ impl Syntax<'parse> for Field {
     }
 }
 
+pub enum ParsedMember {
+    ParsedMethod(ParsedMethod),
+    ParsedField(ParsedField),
+}
+
+/// Represents a parse of something like `foo: Type`
+#[derive(Clone, DebugWith)]
+pub struct ParsedMethod {
+    pub name: Spanned<GlobalIdentifier, FileName>,
+    pub signature: ParsedFunctionDeclaration,
+}
+
+impl LazyParsedEntity for ParsedMethod {
+    fn parse_children(
+        &self,
+        _entity: Entity,
+        _db: &dyn LazyParsedEntityDatabase,
+    ) -> WithError<Seq<ParsedEntity>> {
+        WithError::ok(Seq::default())
+    }
+
+    fn parse_generic_declarations(
+        &self,
+        _entity: Entity,
+        _db: &dyn LazyParsedEntityDatabase,
+    ) -> WithError<Result<Arc<ty::GenericDeclarations>, ErrorReported>> {
+        WithError::ok(Ok(ty::GenericDeclarations::empty(None)))
+    }
+
+    fn parse_type(
+        &self,
+        entity: Entity,
+        db: &dyn LazyParsedEntityDatabase,
+    ) -> WithError<ty::Ty<Declaration>> {
+        self.signature.parse_type(entity, db)
+    }
+
+    fn parse_signature(
+        &self,
+        entity: Entity,
+        db: &dyn LazyParsedEntityDatabase,
+    ) -> WithError<Result<ty::Signature<Declaration>, ErrorReported>> {
+        self.signature.parse_signature(entity, db)
```

Comment:
Ah, this is the problem -- the signature should include the `self` argument. For that to work, we'll need some context about the owner.